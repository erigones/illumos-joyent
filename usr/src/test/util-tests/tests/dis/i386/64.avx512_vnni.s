/*
 * This file and its contents are supplied under the terms of the
 * Common Development and Distribution License ("CDDL"), version 1.0.
 * You may only use this file in accordance with the terms of version
 * 1.0 of the CDDL.
 *
 * A full copy of the text of the CDDL should have accompanied this
 * source.  A copy of the CDDL is also available via the Internet at
 * http://www.illumos.org/license/CDDL.
 */

/*
 * Copyright 2019 Joyent, Inc.
 * Copyright 2024 Oxide Computer Company
 */

/*
 * 64-bit AVX-512 VNNI instruction disassembly.
 */

.text
.align 16
.globl libdis_test
.type libdis_test, @function
libdis_test:
	vpdpbusd	%xmm0, %xmm1, %xmm2
	vpdpbusd	%ymm3, %ymm4, %ymm5
	vpdpbusd	%zmm6, %zmm7, %zmm8
	vpdpbusd	%xmm9, %xmm10, %xmm11{%k1}
	vpdpbusd	%xmm12, %xmm13, %xmm14{%k2}{z}
	vpdpbusd	%ymm15, %ymm16, %ymm17{%k3}
	vpdpbusd	%ymm18, %ymm19, %ymm20{%k4}{z}
	vpdpbusd	%zmm21, %zmm22, %zmm23{%k5}
	vpdpbusd	%zmm24, %zmm25, %zmm26{%k6}{z}
	vpdpbusd	0x64(%rax), %zmm27, %zmm28
	vpdpbusd	0x123456(%rbx, %rcx, 8), %zmm29, %zmm30
	vpdpbusd	(%rdx){1to4}, %xmm1, %xmm2
	vpdpbusd	0x23(%rdx){1to4}, %xmm1, %xmm2
	vpdpbusd	(%rdx){1to8}, %ymm1, %ymm2
	vpdpbusd	0x23(%rdx){1to8}, %ymm1, %ymm2
	vpdpbusd	(%rdx){1to16}, %zmm1, %zmm2
	vpdpbusd	0x23(%rdx){1to16}, %zmm1, %zmm2

	vpdpbusds	%xmm0, %xmm1, %xmm2
	vpdpbusds	%ymm3, %ymm4, %ymm5
	vpdpbusds	%zmm6, %zmm7, %zmm8
	vpdpbusds	%xmm9, %xmm10, %xmm11{%k1}
	vpdpbusds	%xmm12, %xmm13, %xmm14{%k2}{z}
	vpdpbusds	%ymm15, %ymm16, %ymm17{%k3}
	vpdpbusds	%ymm18, %ymm19, %ymm20{%k4}{z}
	vpdpbusds	%zmm21, %zmm22, %zmm23{%k5}
	vpdpbusds	%zmm24, %zmm25, %zmm26{%k6}{z}
	vpdpbusds	0x64(%rax), %zmm27, %zmm28
	vpdpbusds	0x123456(%rbx, %rcx, 8), %zmm29, %zmm30
	vpdpbusds	(%rdx){1to4}, %xmm1, %xmm2
	vpdpbusds	0x23(%rdx){1to4}, %xmm1, %xmm2
	vpdpbusds	(%rdx){1to8}, %ymm1, %ymm2
	vpdpbusds	0x23(%rdx){1to8}, %ymm1, %ymm2
	vpdpbusds	(%rdx){1to16}, %zmm1, %zmm2
	vpdpbusds	0x23(%rdx){1to16}, %zmm1, %zmm2

	vpdpwssd	%xmm0, %xmm1, %xmm2
	vpdpwssd	%ymm3, %ymm4, %ymm5
	vpdpwssd	%zmm6, %zmm7, %zmm8
	vpdpwssd	%xmm9, %xmm10, %xmm11{%k1}
	vpdpwssd	%xmm12, %xmm13, %xmm14{%k2}{z}
	vpdpwssd	%ymm15, %ymm16, %ymm17{%k3}
	vpdpwssd	%ymm18, %ymm19, %ymm20{%k4}{z}
	vpdpwssd	%zmm21, %zmm22, %zmm23{%k5}
	vpdpwssd	%zmm24, %zmm25, %zmm26{%k6}{z}
	vpdpwssd	0x64(%rax), %zmm27, %zmm28
	vpdpwssd	0x123456(%rbx, %rcx, 8), %zmm29, %zmm30
	vpdpwssd	(%rdx){1to4}, %xmm1, %xmm2
	vpdpwssd	0x23(%rdx){1to4}, %xmm1, %xmm2
	vpdpwssd	(%rdx){1to8}, %ymm1, %ymm2
	vpdpwssd	0x23(%rdx){1to8}, %ymm1, %ymm2
	vpdpwssd	(%rdx){1to16}, %zmm1, %zmm2
	vpdpwssd	0x23(%rdx){1to16}, %zmm1, %zmm2

	vpdpwssds	%xmm0, %xmm1, %xmm2
	vpdpwssds	%ymm3, %ymm4, %ymm5
	vpdpwssds	%zmm6, %zmm7, %zmm8
	vpdpwssds	%xmm9, %xmm10, %xmm11{%k1}
	vpdpwssds	%xmm12, %xmm13, %xmm14{%k2}{z}
	vpdpwssds	%ymm15, %ymm16, %ymm17{%k3}
	vpdpwssds	%ymm18, %ymm19, %ymm20{%k4}{z}
	vpdpwssds	%zmm21, %zmm22, %zmm23{%k5}
	vpdpwssds	%zmm24, %zmm25, %zmm26{%k6}{z}
	vpdpwssds	0x64(%rax), %zmm27, %zmm28
	vpdpwssds	0x123456(%rbx, %rcx, 8), %zmm29, %zmm30
	vpdpwssds	(%rdx){1to4}, %xmm1, %xmm2
	vpdpwssds	0x23(%rdx){1to4}, %xmm1, %xmm2
	vpdpwssds	(%rdx){1to8}, %ymm1, %ymm2
	vpdpwssds	0x23(%rdx){1to8}, %ymm1, %ymm2
	vpdpwssds	(%rdx){1to16}, %zmm1, %zmm2
	vpdpwssds	0x23(%rdx){1to16}, %zmm1, %zmm2
.size libdis_test, [.-libdis_test]
