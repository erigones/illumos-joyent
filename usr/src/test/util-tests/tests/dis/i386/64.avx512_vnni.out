    libdis_test:       62 f2 75 08 50 d0  vpdpbusd %xmm0,%xmm1,%xmm2
    libdis_test+0x6:   62 f2 5d 28 50 eb  vpdpbusd %ymm3,%ymm4,%ymm5
    libdis_test+0xc:   62 72 45 48 50 c6  vpdpbusd %zmm6,%zmm7,%zmm8
    libdis_test+0x12:  62 52 2d 09 50 d9  vpdpbusd %xmm9,%xmm10,%xmm11{%k1}
    libdis_test+0x18:  62 52 15 8a 50 f4  vpdpbusd %xmm12,%xmm13,%xmm14{%k2}{z}
    libdis_test+0x1e:  62 c2 7d 23 50 cf  vpdpbusd %ymm15,%ymm16,%ymm17{%k3}
    libdis_test+0x24:  62 a2 65 a4 50 e2  vpdpbusd %ymm18,%ymm19,%ymm20{%k4}{z}
    libdis_test+0x2a:  62 a2 4d 45 50 fd  vpdpbusd %zmm21,%zmm22,%zmm23{%k5}
    libdis_test+0x30:  62 02 35 c6 50 d0  vpdpbusd %zmm24,%zmm25,%zmm26{%k6}{z}
    libdis_test+0x36:  62 62 25 40 50 a0  vpdpbusd 0x64(%rax),%zmm27,%zmm28
                       64 00 00 00 
    libdis_test+0x40:  62 62 15 40 50 b4  vpdpbusd 0x123456(%rbx,%rcx,8),%zmm29,%zmm30
                       cb 56 34 12 00 
    libdis_test+0x4b:  62 f2 75 18 50 12  vpdpbusd (%rdx){1to4},%xmm1,%xmm2
    libdis_test+0x51:  62 f2 75 18 50 92  vpdpbusd 0x23(%rdx){1to4},%xmm1,%xmm2
                       23 00 00 00 
    libdis_test+0x5b:  62 f2 75 38 50 12  vpdpbusd (%rdx){1to8},%ymm1,%ymm2
    libdis_test+0x61:  62 f2 75 38 50 92  vpdpbusd 0x23(%rdx){1to8},%ymm1,%ymm2
                       23 00 00 00 
    libdis_test+0x6b:  62 f2 75 58 50 12  vpdpbusd (%rdx){1to16},%zmm1,%zmm2
    libdis_test+0x71:  62 f2 75 58 50 92  vpdpbusd 0x23(%rdx){1to16},%zmm1,%zmm2
                       23 00 00 00 
    libdis_test+0x7b:  62 f2 75 08 51 d0  vpdpbusds %xmm0,%xmm1,%xmm2
    libdis_test+0x81:  62 f2 5d 28 51 eb  vpdpbusds %ymm3,%ymm4,%ymm5
    libdis_test+0x87:  62 72 45 48 51 c6  vpdpbusds %zmm6,%zmm7,%zmm8
    libdis_test+0x8d:  62 52 2d 09 51 d9  vpdpbusds %xmm9,%xmm10,%xmm11{%k1}
    libdis_test+0x93:  62 52 15 8a 51 f4  vpdpbusds %xmm12,%xmm13,%xmm14{%k2}{z}
    libdis_test+0x99:  62 c2 7d 23 51 cf  vpdpbusds %ymm15,%ymm16,%ymm17{%k3}
    libdis_test+0x9f:  62 a2 65 a4 51 e2  vpdpbusds %ymm18,%ymm19,%ymm20{%k4}{z}
    libdis_test+0xa5:  62 a2 4d 45 51 fd  vpdpbusds %zmm21,%zmm22,%zmm23{%k5}
    libdis_test+0xab:  62 02 35 c6 51 d0  vpdpbusds %zmm24,%zmm25,%zmm26{%k6}{z}
    libdis_test+0xb1:  62 62 25 40 51 a0  vpdpbusds 0x64(%rax),%zmm27,%zmm28
                       64 00 00 00 
    libdis_test+0xbb:  62 62 15 40 51 b4  vpdpbusds 0x123456(%rbx,%rcx,8),%zmm29,%zmm30
                       cb 56 34 12 00 
    libdis_test+0xc6:  62 f2 75 18 51 12  vpdpbusds (%rdx){1to4},%xmm1,%xmm2
    libdis_test+0xcc:  62 f2 75 18 51 92  vpdpbusds 0x23(%rdx){1to4},%xmm1,%xmm2
                       23 00 00 00 
    libdis_test+0xd6:  62 f2 75 38 51 12  vpdpbusds (%rdx){1to8},%ymm1,%ymm2
    libdis_test+0xdc:  62 f2 75 38 51 92  vpdpbusds 0x23(%rdx){1to8},%ymm1,%ymm2
                       23 00 00 00 
    libdis_test+0xe6:  62 f2 75 58 51 12  vpdpbusds (%rdx){1to16},%zmm1,%zmm2
    libdis_test+0xec:  62 f2 75 58 51 92  vpdpbusds 0x23(%rdx){1to16},%zmm1,%zmm2
                       23 00 00 00 
    libdis_test+0xf6:  62 f2 75 08 52 d0  vpdpwssd %xmm0,%xmm1,%xmm2
    libdis_test+0xfc:  62 f2 5d 28 52 eb  vpdpwssd %ymm3,%ymm4,%ymm5
    libdis_test+0x102: 62 72 45 48 52 c6  vpdpwssd %zmm6,%zmm7,%zmm8
    libdis_test+0x108: 62 52 2d 09 52 d9  vpdpwssd %xmm9,%xmm10,%xmm11{%k1}
    libdis_test+0x10e: 62 52 15 8a 52 f4  vpdpwssd %xmm12,%xmm13,%xmm14{%k2}{z}
    libdis_test+0x114: 62 c2 7d 23 52 cf  vpdpwssd %ymm15,%ymm16,%ymm17{%k3}
    libdis_test+0x11a: 62 a2 65 a4 52 e2  vpdpwssd %ymm18,%ymm19,%ymm20{%k4}{z}
    libdis_test+0x120: 62 a2 4d 45 52 fd  vpdpwssd %zmm21,%zmm22,%zmm23{%k5}
    libdis_test+0x126: 62 02 35 c6 52 d0  vpdpwssd %zmm24,%zmm25,%zmm26{%k6}{z}
    libdis_test+0x12c: 62 62 25 40 52 a0  vpdpwssd 0x64(%rax),%zmm27,%zmm28
                       64 00 00 00 
    libdis_test+0x136: 62 62 15 40 52 b4  vpdpwssd 0x123456(%rbx,%rcx,8),%zmm29,%zmm30
                       cb 56 34 12 00 
    libdis_test+0x141: 62 f2 75 18 52 12  vpdpwssd (%rdx){1to4},%xmm1,%xmm2
    libdis_test+0x147: 62 f2 75 18 52 92  vpdpwssd 0x23(%rdx){1to4},%xmm1,%xmm2
                       23 00 00 00 
    libdis_test+0x151: 62 f2 75 38 52 12  vpdpwssd (%rdx){1to8},%ymm1,%ymm2
    libdis_test+0x157: 62 f2 75 38 52 92  vpdpwssd 0x23(%rdx){1to8},%ymm1,%ymm2
                       23 00 00 00 
    libdis_test+0x161: 62 f2 75 58 52 12  vpdpwssd (%rdx){1to16},%zmm1,%zmm2
    libdis_test+0x167: 62 f2 75 58 52 92  vpdpwssd 0x23(%rdx){1to16},%zmm1,%zmm2
                       23 00 00 00 
    libdis_test+0x171: 62 f2 75 08 53 d0  vpdpwssds %xmm0,%xmm1,%xmm2
    libdis_test+0x177: 62 f2 5d 28 53 eb  vpdpwssds %ymm3,%ymm4,%ymm5
    libdis_test+0x17d: 62 72 45 48 53 c6  vpdpwssds %zmm6,%zmm7,%zmm8
    libdis_test+0x183: 62 52 2d 09 53 d9  vpdpwssds %xmm9,%xmm10,%xmm11{%k1}
    libdis_test+0x189: 62 52 15 8a 53 f4  vpdpwssds %xmm12,%xmm13,%xmm14{%k2}{z}
    libdis_test+0x18f: 62 c2 7d 23 53 cf  vpdpwssds %ymm15,%ymm16,%ymm17{%k3}
    libdis_test+0x195: 62 a2 65 a4 53 e2  vpdpwssds %ymm18,%ymm19,%ymm20{%k4}{z}
    libdis_test+0x19b: 62 a2 4d 45 53 fd  vpdpwssds %zmm21,%zmm22,%zmm23{%k5}
    libdis_test+0x1a1: 62 02 35 c6 53 d0  vpdpwssds %zmm24,%zmm25,%zmm26{%k6}{z}
    libdis_test+0x1a7: 62 62 25 40 53 a0  vpdpwssds 0x64(%rax),%zmm27,%zmm28
                       64 00 00 00 
    libdis_test+0x1b1: 62 62 15 40 53 b4  vpdpwssds 0x123456(%rbx,%rcx,8),%zmm29,%zmm30
                       cb 56 34 12 00 
    libdis_test+0x1bc: 62 f2 75 18 53 12  vpdpwssds (%rdx){1to4},%xmm1,%xmm2
    libdis_test+0x1c2: 62 f2 75 18 53 92  vpdpwssds 0x23(%rdx){1to4},%xmm1,%xmm2
                       23 00 00 00 
    libdis_test+0x1cc: 62 f2 75 38 53 12  vpdpwssds (%rdx){1to8},%ymm1,%ymm2
    libdis_test+0x1d2: 62 f2 75 38 53 92  vpdpwssds 0x23(%rdx){1to8},%ymm1,%ymm2
                       23 00 00 00 
    libdis_test+0x1dc: 62 f2 75 58 53 12  vpdpwssds (%rdx){1to16},%zmm1,%zmm2
    libdis_test+0x1e2: 62 f2 75 58 53 92  vpdpwssds 0x23(%rdx){1to16},%zmm1,%zmm2
                       23 00 00 00 
